### PR TITLE
Fixed theme in `Choose account for ODK Collect` dialog

### LINKED ISSUE #3729: Dark `Choose account for ODK Collect` dialog for light mode
#### Software and hardware versions 
Collect v1.24.0
Older versions weren't verified

#### Problem description
Dark `Choose account for ODK Collect` dialog for light mode

#### Steps to reproduce the problem
1. User uses the light mode
2. User goes to `General Settings` -> `Server`
3. User selects Google Drive in Type option
4. User clicks on `Google account` to open dialog `Choose account for ODK Collect`
5. **Dialog is dark** 

Screenshot_20200320-163008

#### Expected behavior
Dialog `Choose account for ODK Collect` is light for light mode and dark for dark mode

### PR DESCRIPTION
**Pull request by SaumiaSinghal**
Closes #3729 

#### What has been done to verify that this works as intended?
I tested it.
Light Mode           |     Dark Mode
:-------------------------:|:-------------------------:
WhatsApp Image 2020-03-25 at 01 40 52 (1) |  WhatsApp Image 2020-03-25 at 01 40 52

#### Why is this the best possible solution? Were any other approaches considered?
I used `AccountManager.newAccountChooserIntent()` to create the intent when the theme is light, and `AccountPicker.newAccountChooserIntent()` when theme of the app is dark. I found that the default theme of the account chooser intent is light, when created by AccountManager. 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
No

#### Do we need any specific form for testing your changes? If so, please attach one.
No

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java [hunk 1]
```diff
@@ -89,8 +92,20 @@ private void initCredential(@NonNull Context context) {
                 .usingOAuth2(context, Collections.singletonList(DriveScopes.DRIVE))
                 .setBackOff(new ExponentialBackOff());
 
-        intentChooseAccount = credential.newChooseAccountIntent();
         themeUtils = new ThemeUtils(context);
+
+        if (themeUtils.getAccountPickerTheme() == 1) {
+            intentChooseAccount = AccountManager.newChooseAccountIntent(null,
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Do these two methods definitely do the same thing? I'm a little worried about using a whole different integration based on the theme.

Comment (SaumiaSinghal) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Hi @seadowg! Thanks for reviewing. `AccountPicker` is a part of Google Play services. I went through the docs of [`AccountPicker.newChooseAccountIntent()`](https://developers.google.com/android/reference/com/google/android/gms/common/AccountPicker#newChooseAccountIntent(com.google.android.gms.common.AccountPicker.AccountChooserOptions)) and [`AccountManager.newChooseAccountIntent()`](https://developer.android.com/reference/android/accounts/AccountManager#newChooseAccountIntent(android.accounts.Account,%2520java.util.ArrayList%253Candroid.accounts.Account%253E,%2520java.lang.String%255B%255D,%2520boolean,%2520java.lang.String,%2520java.lang.String,%2520java.lang.String%255B%255D,%2520android.os.Bundle)), and found that both take the same arguments, and do the same thing.

Comment (SaumiaSinghal) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Using a different integration just to fix the theme, seems a little strange to me. But I searched, and I was not able to change the theme in either of the cases with `AccountPicker` or `AccountManager`, by overriding variables in the intent.

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Hmmm I'm not sure about the ramifications of this but I'm not really comfortable with it without doing some research. @grzesiek2010 do you have an opinion on this?

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: For me it doesn't work and I also don't understand the trick here. If I go to the body of `newChooseAccountIntent()`
it returns:
`    return AccountPicker.newChooseAccountIntent(selectedAccount,
        null,
        new String[] {GoogleAccountManager.ACCOUNT_TYPE},
        true,
        null,
        null,
        null,
        null);`
so generally the same parameters so where is the fix?

Comment (SaumiaSinghal) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Hi @grzesiek2010! Thanks for reviewing. I could not find much in docs about the theme. But I was not able to change the theme by overriding in intent, in both cases. 
I agree that both [`AccountPicker.newChooseAccountIntent()`](https://developers.google.com/android/reference/com/google/android/gms/common/AccountPicker#newChooseAccountIntent(com.google.android.gms.common.AccountPicker.AccountChooserOptions)) and [`AccountManager.newChooseAccountIntent()`](https://developer.android.com/reference/android/accounts/AccountManager#newChooseAccountIntent(android.accounts.Account,%2520java.util.ArrayList%253Candroid.accounts.Account%253E,%2520java.lang.String%255B%255D,%2520boolean,%2520java.lang.String,%2520java.lang.String,%2520java.lang.String%255B%255D,%2520android.os.Bundle)) take the same parameters, but the the default theme in both cases is different, so I decided to create intent in differnet ways, for light and dark theme.

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: > but the the default theme in both cases is different

I can't see it on my android 5, I mean it's still dark always.

Comment (SaumiaSinghal) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: I tried testing on the emulator. I found the same problem you mentioned on android 5. However, I didn't find it on Android 5.1, 6 and on my device running Android 9.

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: We shouldn't merge such a fix if it doesn't wok on all devices and is kind of suspicious.

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java:98]: Ok so I'm closing. @SaumiaSinghal thanks for trying!

**Pull request closed without merging**